tool/site/client/src/pages/withdrawPage.tsx

```diff
@@ -37,41 +37,55 @@ const WithdrawPage: React.FC<WithdrawPageProps> = ({
   return (
     <LongRunningTask<DispenseNotificationTaskResponse> task={retrieveDispenseNotificationsTask} loadingMessage="Retrieving dispense notifications.">
       {taskResponse => {
+        const dispenseEvents = createPrescriptionDispenseEvents(taskResponse.dispenseNotifications)
+        const lastDispenseId = dispenseEvents.length > 0
+          ? dispenseEvents[dispenseEvents.length - 1].dispenseEventId
+          : undefined
+        const heading = lastDispenseId
+          ? `Withdrawing Dispense: ${lastDispenseId}`
+          : "Withdraw Unavailable"
         if (!withdrawFormValues) {
           return (
             <>
-              <Label isPageHeading>Withdraw prescription</Label>
-              <DispenseNotificationsTable dispenseNotifications={taskResponse.dispenseNotifications}/>
+              <Label isPageHeading>{heading}</Label>
+              <DispenseEventTable events={dispenseEvents} prescriptionId={prescriptionId}/>
               <WithdrawForm prescriptionId={prescriptionId} onSubmit={setWithdrawFormValues}/>
             </>
           )
         }
         const sendWithdrawTask = () => sendWithdraw(baseUrl, prescriptionId, withdrawFormValues)
```

Comment:
we can pass previous dispenses into this method to avoid having to get them again from the backend

Reply:
done